src/main/java/dk/kb/labsapi/ImageExport.java

```diff
@@ -193,58 +267,73 @@ public int setUsableEndTime(int endTime){
 
     /**
-     * Parse Solr QueryResponse of newspaper pages into list of individual illustrations.
-     * @param solrResponse to extract illustrations from.
-     * @return a list of strings. Each string contains metadata for a single illustration from the input query response.
+     * Get metadata values for all pages from query.
+     * The returned list contains an object of metadata from each single page.
+     * @return a list of objects consisting of pageUUID, pageWidth and pageHeight values for each page in the response from solr.
      */
-     private List<String> getIllustrationsList(QueryResponse solrResponse) {
-        SolrDocumentList responseList = new SolrDocumentList();
-        responseList = solrResponse.getResults();
-        List<String> illustrationList = new ArrayList<>();
-
-        // Set for deduplication on pageUUID level
-        Set<String> uniqueUUIDs = new HashSet<>();
-
-        // Extract metadata from documents in solr response
-        for (int i = 0; i<responseList.size(); i++) {
-            String pageUUID = responseList.get(i).getFieldValue("pageUUID").toString();
-            if (!uniqueUUIDs.add(pageUUID)){
-                continue;
-            }
-            long pageWidth = (long) responseList.get(i).getFieldValue("page_width");
-            long pageHeight = (long) responseList.get(i).getFieldValue("page_height");
-            List<String> illustrations = (List<String>) responseList.get(i).getFieldValue("illustration");
-            // Check if illustrations are present. If not, continue to next SolrDocument in list
-            if (illustrations == null) {
-                continue;
-            }
-            // Create metadata string for each illustration
-            for (int j = 0; j< illustrations.size(); j++){
-                illustrations.set(j, illustrations.get(j) + "," + pageUUID + "," + pageWidth + "," + pageHeight);
-            }
-            illustrationList.addAll(illustrations);
+    public List<FullPageMetadata> getMetadataForFullPage(QueryResponse solrResponse) {
+         List<FullPageMetadata> pages = new ArrayList<>();
 
-        }
-        return illustrationList;
+         SolrDocumentList documents = solrResponse.getResults();
+         for (SolrDocument doc : documents){
+             FullPageMetadata page = new FullPageMetadata(doc.get("pageUUID").toString(), (Long) doc.get("page_width"), (Long) doc.get("page_height"));
+             pages.add(page);
+         }
+
+         return pages;
+    }
+
+    /**
+     * Remove prefix <em>doms_aviser_page:uuid</em> from pageUUID.
+     * @param pageUUID to convert.
+     * @return correct pageUUID without prefix.
+     */
+    private String convertPageUUID(String pageUUID){
+         String correctUUID = "";
+         final Pattern pagePattern = Pattern.compile("doms_aviser_page:uuid:(\\S*)");
+         Matcher m = pagePattern.matcher(pageUUID);
+         if (m.matches()){
+             correctUUID = m.group(1);
+         }
```

Comment:
So what happens if there are no match? Is that a common and non-problematic occurrence or a sign of something unexpected going wrong?Should there be some logging or exception here?

Reply:
Please log the problematic input String when logging to help with debugging.